cmd/auto-updater/main.go

```diff
@@ -132,11 +149,42 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
 			}
 			log.Println("finished downloading", downloadURL, "to", tempArchive.Name())
 
-			// TODO next PRs
-			// * rename current binary into `storagenode.old.<release>.exe`.
-			// * unzip downloaded binary in place of current binary
-			// * compare extracted binary with version from suggested version
-			// * restart storage node service
+			extension := filepath.Ext(binaryLocation)
+			if extension != "" {
+				extension = "." + extension
+			}
+
+			dir, _ := filepath.Split(binaryLocation)
+			backupExec := filepath.Join(dir, "storagenode.old."+currentVersion.String()+extension)
+
+			if err = os.Rename(binaryLocation, backupExec); err != nil {
+				return err
+			}
+
+			err = unpackBinary(ctx, tempArchive.Name(), binaryLocation)
+			if err != nil {
+				return err
+			}
+
+			newVersion, err := binaryVersion(binaryLocation)
+			if err != nil {
+				return err
+			}
+
+			if suggestedVersion.Compare(newVersion) != 0 {
+				return errs.New("invalid version downloaded: wants %s got %s", suggestedVersion.String(), newVersion.String())
+			}
+
+			log.Println("restarting service", snServiceName)
+			err = restartSNService(snServiceName)
+			if err != nil {
+				return errs.New("unable to restart service: %v", err)
+			}
+			log.Println("service", snServiceName, "restarted successfully")
+
+			// TODO remove old binary ??
```

Comment:
I'd leave it in case there is any problem it will be easier for an SNO to have the previous binary that it was running.